model/simulationtests/autosize_hvac.rb

```diff
@@ -81,17 +81,96 @@ def curve_biquadratic(model, c_1constant, c_2x, c_3xPOW2, c_4y, c_5yPOW2, c_6xTI
   curve.setCoefficient3xPOW2(c_3xPOW2)
   curve.setMinimumValueofx(minx)
   curve.setMaximumValueofx(maxx)
-  curve.setMinimumCurveOutput(miny)
-  curve.setMaximumCurveOutput(maxy)
+  curve.setMinimumCurveOutput(miny) if !miny.nil?
+  curve.setMaximumCurveOutput(maxy) if !maxy.nil?
   return curve
 end
+
+def make_tes_coil(model)
+  tes_coil = OpenStudio::Model::CoilCoolingDXSingleSpeedThermalStorage.new(model)
+  tes_op_schedule = OpenStudio::Model::ScheduleConstant.new(model)
+  tes_op_schedule.setValue(2) # 2 = Cooling Only Mode
+  tes_coil.setOperationModeControlSchedule(tes_op_schedule)
+  tes_coil.setStorageType('Ice')
+  tes_coil.autosizeFluidStorageVolume # Only autosized if Storage Type is Water
+  tes_coil.autosizeIceStorageCapacity
+  tes_coil.autosizeRatedEvaporatorAirFlowRate
+
+  # Obviously, don't use this in production code. We just assign all curves to
+  # a constant value to avoid E+ crashing on us, while not having to actually
+  # create 41 curves with potentially lots of values...
+  constant_univariate = OpenStudio::Model::CurveCubic.new(model)
+  constant_univariate.setName('Constant Univariate Curve')
+  constant_univariate.setCoefficient1Constant(1.0)
+  constant_univariate.setCoefficient4xPOW3(0.0)
+  constant_univariate.setMinimumValueofx(-100.0)
+  constant_univariate.setMaximumValueofx(100.0)
+
+  constant_bivariate = OpenStudio::Model::CurveBiquadratic.new(model)
+  constant_bivariate.setName('Constant Bivariate Curve')
+  constant_bivariate.setCoefficient1Constant(1.0)
+  constant_bivariate.setMinimumValueofx(-100.0)
+  constant_bivariate.setMaximumValueofx(100.0)
+  constant_bivariate.setMinimumValueofy(-100.0)
+  constant_bivariate.setMaximumValueofy(100.0)
+
+  constant_trivariate = OpenStudio::Model::CurveTriquadratic.new(model)
+  constant_trivariate.setCoefficient1Constant(1.0)
+  constant_trivariate.setMinimumValueofx(-100.0)
+  constant_trivariate.setMaximumValueofx(100.0)
+  constant_trivariate.setMinimumValueofy(-100.0)
+  constant_trivariate.setMaximumValueofy(100.0)
+  constant_trivariate.setMinimumValueofz(-100.0)
+  constant_trivariate.setMaximumValueofz(100.0)
+
+  object_list_to_curve = {
+    'UnivariateFunctions' => constant_univariate,
+    'BivariateFunctions' => constant_bivariate,
+    'TrivariateFunctions' => constant_trivariate
+  }
+
+  # Scan the curve fields, assign a constant curve with the right number of
+  # independent variables
+  iddObject = tes_coil.iddObject
+  iddObject.numFields.times do |i|
+    field = iddObject.getField(i).get
+    field_name = field.name
+    next unless field_name.include?('Curve')
+
+    props = field.properties
+    olist = props.objectLists.first
+    setter = 'set' + field_name.gsub(' ', '')
+    raise "Undefined method #{setter} for field_name=#{field_name} at index #{i}" unless tes_coil.respond_to?(setter)
+
+    tes_coil.send(setter, object_list_to_curve[olist])
+  end
+
+  tes_coil.setCondenserType('EvaporativelyCooled')
```

Comment:
Yeah, I know. Not great as a far as readability goes, but the point of this test is to check the autosized SQL calls, not making a clean model